frex/utils/constraint_type.py

```diff
@@ -0,0 +1,26 @@
+from enum import Enum
+
+
+class ConstraintType(Enum):
+    """
+    ConstraintType is a callable Enum for use in the integer programming solver.
```

Comment:
Interesting choice. You can make the @staticmethods private so they're not exposed.

Enumerators usually have to be value-equatable (like scalars or data structures containing scalars etc.), and methods aren't.

Don't need to change this, though. What you've got is something like a flyweight pattern.